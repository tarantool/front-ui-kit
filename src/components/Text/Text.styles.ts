import { css } from '@emotion/css';

import { baseFontFamily, colors, monoFontFamily } from '../../variables';

export const textStyles = {
  h1: css`
    margin: 0;
    font-family: ${baseFontFamily};
    font-size: 24px;
    line-height: 40px;
    font-weight: 600;
    color: ${colors.dark};
    -webkit-font-smoothing: antialiased;
  `,
  h2: css`
    margin: 0;
    font-family: ${baseFontFamily};
    font-size: 20px;
    line-height: 28px;
    font-weight: 600;
    color: ${colors.dark};
    -webkit-font-smoothing: antialiased;
  `,
  h3: css`
    margin: 0;
    font-family: ${baseFontFamily};
    font-size: 16px;
    line-height: 24px;
    font-weight: 600;
    color: ${colors.dark};
    -webkit-font-smoothing: antialiased;
  `,
  h4: css`
    margin: 0;
    font-family: ${baseFontFamily};
    font-size: 14px;
    line-height: 22px;
    font-weight: 600;
    color: ${colors.dark};
    -webkit-font-smoothing: antialiased;
  `,
  h5: css`
    margin: 0;
    font-family: ${baseFontFamily};
    font-size: 12px;
    line-height: 22px;
    text-transform: uppercase;
    letter-spacing: 0.01em;
    color: ${colors.dark};
    font-weight: 600;
    -webkit-font-smoothing: antialiased;
  `,
  p: css`
    margin: 0;
    font-family: ${baseFontFamily};
    font-size: 12px;
    line-height: 20px;
    color: #000000;
    -webkit-font-smoothing: antialiased;
  `,
  code: css`
    margin: 0;
    font-family: ${monoFontFamily};
    font-size: 14px;
    line-height: 20px;
    color: #000000;
    -webkit-font-smoothing: antialiased;
    b {
      font-weight: 600;
    }
  `,
  basic: css`
    margin: 0;
    font-family: ${baseFontFamily};
    font-size: 14px;
    line-height: 22px;
    color: #000000;
    -webkit-font-smoothing: antialiased;
    b {
      font-weight: 600;
    }
  `,
  upperCase: css`
    text-transform: uppercase;
  `,
  noCase: css`
    text-transform: none;
  `,
};
