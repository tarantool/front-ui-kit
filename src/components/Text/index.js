// @flow
import * as React from 'react'
import { css, cx } from 'emotion'

export const textStyles = {
  h1: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 24px;
    line-height: 38px;
    font-weight: 600;
    letter-spacing: 0.48px;
    text-transform: uppercase;
    color: #000000;
  `,
  h2: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 20px;
    line-height: 28px;
    font-weight: 600;
    letter-spacing: 0.4px;
    text-transform: uppercase;
    color: #000000;
  `,
  h3: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 16px;
    line-height: 24px;
    font-weight: 600;
    letter-spacing: 0.32px;
    text-transform: uppercase;
    color: #cf1322;
  `,
  h4: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 16px;
    line-height: 24px;
    font-weight: 600;
    letter-spacing: 0.32px;
    color: #000000;
  `,
  h5: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 12px;
    line-height: 22px;
    letter-spacing: 0.24px;
    color: #000000;
  `,
  p: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 12px;
    line-height: 20px;
    letter-spacing: 0.28px;
    color: #000000;
  `,
  code: css`
    margin: 0;
    font-family: 'Monaco', monospace;
    font-size: 14px;
    line-height: 22px;
    color: #000000;
    b {
      font-weight: 600;
    }
  `,
  basic: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 14px;
    line-height: 22px;
    color: #000000;
    b {
      font-weight: 600;
    }
  `,
  upperCase: css`
    text-transform: uppercase;
  `,
  noCase: css`
    text-transform: none;
  `
};

type TextProps = {
  className?: string,
  children?: React.Node,
  tag?: string,
  upperCase?: boolean,
  noCase?: boolean,
  title?: string,
  variant?:
    |'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'p' | 'basic' | 'code',
  // colorVariant?: 'inherit' | 'primary' | 'secondary' | 'disabled' | 'white' | 'error',
};

export const Text = ({
  className,
  children,
  tag,
  upperCase,
  noCase,
  title,
  variant = 'basic'
}:
TextProps) => {
  const Element = tag || (variant === 'basic' ? 'span' : variant);

  return (
    <Element
      className={cx(
        textStyles[variant],
        {
          [textStyles.upperCase]: upperCase,
          [textStyles.noCase]: noCase
        },
        className
      )}
      title={title}
    >
      {children}
    </Element>
  );
};
