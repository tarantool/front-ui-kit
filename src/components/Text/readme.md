```
<Text variant='h1'>Heading 1</Text>
<Text variant='h2'>Heading 2</Text>
<Text variant='h3'>Heading 3</Text>
<Text variant='h4'>Heading 4</Text>
<Text variant='h5'>Heading 5</Text>
<Text variant='p'>paragraph</Text>
<Text>basic</Text>
<br />
<Text variant='h3' tag='span'>H3 styled span</Text>
<Text variant='h4' tag='span'>H4 styled span</Text>
```
