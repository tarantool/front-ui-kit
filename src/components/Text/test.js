import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from './index';

it('Text renders correctly', () => {
  const tree = renderer.create(
    <>
      <Text variant='h1' title='Text title'>Heading 1 - 24px, 38px, 600, spacing 0.48px</Text>
      <Text variant='h2'>Heading 2 - 20px, 28px, 600, spacing 0.4px</Text>
      <Text variant='h3'>Heading 3 - 16px, 24px, 600, spacing 0.32px</Text>
      <Text variant='h4'>Heading 4 - 16px, 24px, 600, spacing 0.32px</Text>
      <Text variant='h5'>Heading 5 - 12px, 22px, 400, spacing 0.24px</Text>
      <Text variant='p'>paragraph - 12px, 20px, 400, spacing: 0.28px</Text>
      <Text>basic - 14px, 22px, 400</Text>
      <Text variant='code'>code - 14px, 22px, 400</Text>
      <Text variant='h3' tag='span'>H3 styled span</Text>
      <Text variant='h4' tag='span'>H4 styled span</Text>
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
