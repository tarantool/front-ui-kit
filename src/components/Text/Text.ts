import { MouseEvent, ReactNode, createElement, forwardRef } from 'react';
import { cx } from '@emotion/css';

import { textStyles } from './Text.styles';

export { textStyles };

export type TextStyleVariants = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'p' | 'basic' | 'code';

export interface TextProps {
  className?: string;
  htmlFor?: string;
  children?: ReactNode;
  tag?: string;
  href?: string;
  upperCase?: boolean;
  noCase?: boolean;
  title?: string;
  variant?: TextStyleVariants;
  onClick?: (event: MouseEvent<HTMLElement>) => void;
  // colorVariant?: 'inherit' | 'primary' | 'secondary' | 'disabled' | 'white' | 'error',
}

export const Text = forwardRef<HTMLElement, TextProps>(
  ({ className, children, tag, upperCase, noCase, variant = 'basic', ...props }, ref) => {
    const Element = tag || (variant === 'basic' ? 'span' : variant);

    return createElement(
      Element,
      {
        ...props,
        className: cx(
          textStyles[variant],
          {
            [textStyles.upperCase]: upperCase,
            [textStyles.noCase]: noCase,
          },
          className
        ),
        ref,
      },
      children
    );
  }
);
