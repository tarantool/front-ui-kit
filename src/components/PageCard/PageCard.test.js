import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from '../Text';
import { PageCard } from './index';

it('PageCard renders with close button', () => {
  const tree = renderer.create(
    <>
      <PageCard title="Bootstrap vshard" onClose={() => console.log('Close clicked')}>
        <Text variant='h4'>When you finish edition topology. To render storages operable.</Text>
      </PageCard>
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('PageCard renders without close button', () => {
  const tree = renderer.create(
    <>
      <PageCard title="Bootstrap vshard">
        <Text>Afterwards, any change in topology will trigger data rebalancing</Text>
      </PageCard>
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
