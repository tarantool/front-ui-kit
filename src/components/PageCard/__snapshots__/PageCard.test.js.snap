// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCard renders with close button 1`] = `
<div
  className="css-pb1zll"
>
  <h2
    className="css-19t58j"
  >
    Bootstrap vshard
  </h2>
  <svg
    className="css-14qwf54"
    onClick={[Function]}
    viewBox="0 0 16 16"
  >
    <use
      xlinkHref="#close"
    />
  </svg>
  <div>
    <h4
      className="css-115iv0u"
    >
      When you finish edition topology. To render storages operable.
    </h4>
  </div>
</div>
`;

exports[`PageCard renders without close button 1`] = `
<div
  className="css-pb1zll"
>
  <h2
    className="css-19t58j"
  >
    Bootstrap vshard
  </h2>
  <div>
    <span
      className="css-1ggp6x0"
    >
      Afterwards, any change in topology will trigger data rebalancing
    </span>
  </div>
</div>
`;
